### ISSUE TITLE
Best Practices for Domain ID / Prompt Initialization When Fine-Tuning on Similar Domains

### ISSUE THREAD

**Issue by gianlucageraci** [2toinf/X-VLA#64]
Dear authors,

thanks very much for your great work on X-VLA and making the code publicly available. I have a question regarding domain id usage when finetuning.
In your paper, you specify that two-step adaptation (i.e. Prompt Warm-Up and then Joint Policy Adaptation) is the recommended recipe for specializing a model to a target domain. You write 
> To deploy this model on novel domains with new hardware configurations ℎ_new, we propose a
lightweight two-step adaptation procedure:
> (1) Prompt warm-up. We introduce new sets of learnable prompt 𝑝_new ∈ ℝ^𝑘 for ℎ_new. The prompt is first
warmed up while keeping the pretrained weights frozen. By doing so, prompts are projected to exploit
pretrained embodiment-agnostic features, offering good starts for next-round joint training.
> (2) Joint policy adaptation. Then, we jointly optimize both the backbone and the warmed-up prompt,
jointly adapt to new domains. This two-stage process first lets 𝑝_new encode the hardware-specific setups
of ℎnew, and then finetunes the full policy for effective specialization, sharing the same philosophy used
to adapt LLMs to VLMs.

I understand that if my target domain is very different from the domains included in pre-training, i can simply randomly initialize the learnable prompts and then run the two-step procedure.
My question is now what you recommend to do when my target domain is very similar to a pre-training domain (e.g. DROID). Is it recommended to still follow the two-step approach with random prompt initialization? 
Or is it better to use the prompt correpsonding to the most similar pre-training domain as a starting point for the warm-up? Or would you even recommend to not train the soft prompt at all and keep it frozen if the two domains are similar enough?

Thank you very much for your response!

**Comment by 2toinf** [comment]
Hey! Thanks for your interest in X-VLA.
We have conducted experiments to study the coherence of soft prompts learned from closely related domains. Our results suggest that when the evaluation environment is highly similar to the pretraining domain, inheriting the existing soft prompt is consistently the best choice. We further recommend continuing to fine-tune the inherited prompt to achieve optimal performance.

**Comment by gianlucageraci** [comment]
Hey, thank you very much for your response. It made it a lot clearer for me how to best proceed with training X-VLA 👍 
(i will therefore close the issue)
